### DISCUSSION TITLE
Getting Started run RiscvTimingSimpleCPU or any other RISCV in 32 Bit?

### DISCUSSION THREAD

**Discussion by filipcoja** [gem5/gem5#513] [Q&A] [answered]
Couldnt find anything, how can i get the model to run on 32 bit?

I'm getting this when trying to execute a 32 bit binary:

src/arch/riscv/process.cc:122: fatal: fatal condition isa->rvType() != RV32 occurred: RISC V CPU should run in 32 bits mode

**Comment by rogerchang23424** [comment]
Can you let me know what config you used to execute 32 bits binaries?

You need to set the parameter `riscv_type` of RiscvISA to "RV32".

**Reply by filipcoja** [reply]
I used the riscv_simple.py in the examples folder. 

I know i need to set the riscv_type of riscvisa to rv32 but how can i do it with the CLI? currently im editing the RiscvISA.py file and just rebuild everything like described here:


https://stackoverflow.com/questions/52023721/build-gem5-in-32-bit-mode/77385130#77385130

**Reply by rogerchang23424** [reply, marked as answer]
> https://stackoverflow.com/questions/52023721/build-gem5-in-32-bit-mode/77385130#77385130

The user should not change the default parameter of riscv_type to "RV32". If the config are under example/gem5_library, you can manually add section following the processor=SimpleProcessor(...) to set riscv mode to 32 bits for both SE and FS mode.
```
if args.riscv_32bits:
    for simple_core in processor.cores:
        for i in range(len(simple_core.core.isa)):
            simple_core.core.isa[i].riscv_type = "RV32"
```

Reference: https://github.com/gem5/gem5/blob/06bf783a8533fb1f78c041d2d49492b53eed26b4/tests/gem5/asmtest/configs/riscv_asmtest.py
